### ISSUE TITLE
现存问题

### ISSUE THREAD

**Issue by PIut02** [PIut02/ROG-Zephyrus-G14-GA401-Hackintosh#1]
- [x] 触摸板中断模式
- [x] 睡眠唤醒
- [ ] 睡眠时键盘背光不会完全关闭
- [ ] 扬声器输出
- [ ] 键盘背光控制
- [ ] fn快捷键
- [ ] 亮度表
- [ ] 续航时间很短

**Comment by PIut02** [comment]
在睡眠前手动关闭键盘背光可以避免睡眠时键盘背光无法关闭问题

**Comment by throwaway29292** [comment]
I believe, that the battery life is so short (seems to last only for 1-2 hours on my g14 2020) because the nvidia gpu stays on and isn't switched into some sort of power saving mode. I wonder if there's a way to access the power consumption and frequency of the gpu.

**Comment by PIut02** [comment]
> I believe, that the battery life is so short (seems to last only for 1-2 hours on my g14 2020) because the nvidia gpu stays on and isn't switched into some sort of power saving mode. I wonder if there's a way to access the power consumption and frequency of the gpu.

I tried removing -wegnoegpu in boot-args and found that it was indeed the case, the ssdt patch didn't take effect for some reason, I will remake the patch in the next few days.

**Comment by throwaway29292** [comment]
Wow, thank you. I'll update the efi on my notebook and i'll let you know.

**Comment by throwaway29292** [comment]
The battery life seems to have improved, but it still lasts only around 3-4 hours.

Is there a way, how i can check for actual power draw? The AMD power gadget says 5W idle, but i'm pretty sure, that the total power draw is far bigger.

**Comment by PIut02** [comment]
I don't think so, because the CPU's TDP is only 35W, and when using some large software such as Xcode, the power consumption can reach 30W. The CPU power consumption displayed by AMD Power Gadget is accurate, and I don't know if other software can read AMD's power consumption and temperature.

**Comment by edwin170** [comment]
> I don't think so, because the CPU's TDP is only 35W, and when using some large software such as Xcode, the power consumption can reach 30W. The CPU power consumption displayed by AMD Power Gadget is accurate, and I don't know if other software can read AMD's power consumption and temperature.

do you mean the nvidia gpu reaches 30W ? because shouldn't be the amd graphics as it is the one that we use (nvidia graphics doesn't work on hackintosh)

**Comment by PIut02** [comment]
> do you mean the nvidia gpu reaches 30W ? because shouldn't be the amd graphics as it is the one that we use (nvidia graphics doesn't work on hackintosh)

I mean CPU's power.

**Comment by edwin170** [comment]
oh got it bro.
